spidermon/contrib/validation/cerberus/validator.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+from collections.abc import Mapping
+
+from cerberus.validator import (
+    Validator as cerberus_validator,
+    DocumentError,
+    SchemaError,
+)
+from spidermon.contrib.validation.validator import Validator
+from .translator import CerberusMessageTranslator
+
+
+class CerberusValidator(Validator):
+    default_translator = CerberusMessageTranslator()
+    name = "Cerberus"
+
+    def __init__(self, schema, translator=None, use_default_translator=True):
+        super(CerberusValidator, self).__init__(
+            translator=translator, use_default_translator=use_default_translator
+        )
+        if schema is not None:
+            self._schema = schema
+        elif self._schema is None:
+            if isinstance(self._schema, Mapping):
+                self._schema = {}
+            else:
+                raise SchemaError("Validation schema missing")
+
+    def _validate(self, data, strict=False):
+        if data is None:
+            raise DocumentError("Data is missing")
+        try:
+            validator = cerberus_validator(self._schema)
+            validator.validate(data)
```

Comment:
Only `validator.validate(data)` would raise a `SchemaError`. Don't put too many code inside try..except.

Reply:
Yep, that's true. Will get other lines out of there!

Reply:
On testing again, this also produces SchemaError, I think  that's why it was in the block in the first place, `validator = cerberus_validator(self._schema)`

Removed the error loop though,